authors/apps/authentication/tests/test_jwt.py

```diff
@@ -0,0 +1,173 @@
+from django.test import TestCase
+from django.urls import reverse
+
+from rest_framework import status
+from rest_framework.test import APIClient, APIRequestFactory
+
+from authors.apps.authentication.models import User
+from authors.apps.authentication.backends import JWTAuthentication
+
+
+class JWTAuthenticationTest(TestCase):
+    """ Test the JWT Authentication implementation """
+    def setUp(self):
+        self.user_data = {
+            'user': {
+                'email': 'user1@mail.com',
+                'username': 'user1',
+                'password': 'password'
+                }
+            }
+        self.user = User.objects.create(
+            username='user',
+            email='user@mail.com',
+            password='password'
+        )
+        self.login_data = {
+            'user': {
+                'email': 'user1@mail.com',
+                'password': 'password'
+            }
+        }
+        self.user_token = self.user.token
+        self.user.save()
+        self.client = APIClient()
+
+    def test_user_gets_a_token_when_they_sign_up(self):
+        """ Users should get a token when they successfully log in """
+        response = self.client.post(
+            reverse('authentication:register'),
+            self.user_data,
+            format='json'
+            )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertIn('token', response.data)
+
+    def test_user_gets_a_token_when_they_log_in(self):
+        """ Users should get a token when they successfully log in """
+        self.client.post(
```

Comment:
Why not just use the already registered user and skip the registration step here?